### ISSUE TITLE
Type-only import that merges with namespace cannot be used in a type position.

### ISSUE THREAD

**Issue by rbuckton** [microsoft/TypeScript#36397]
**TypeScript Version:**  3.8.0-dev.20200123

**Search Terms:** type only import merged namesapce

**Code**

```ts
// file1.ts
export interface Foo { }
export namespace Foo {
  export const x = 1;
}

// file2.ts
import type { Foo } from "./file1";

declare function f(): Foo; // (1)
```

**Expected behavior:**

1. No error

**Actual behavior:**

1. Error: `Cannot use namespace 'Foo' as a type.`

**Playground Link:** 
**Related Issues:**

**Comment by likern** [comment]
I'd like to add some additional thoughts to this issue as they might be relevant.

I've found that even though modules are now preffered way to compose code over namespaces, there is still one feature, which doesn't allow to completely replace them.

It is an ability to structure types. In almost any application there is **kinds of types**.

1. Types for modeling data (`Tag`)
2. Types for modeling component properties in React (`TagProps`)
3. Types for modeling callback data and functions - events (`TagEvents`)

If using only modules it's not possible to structure these kinds of types and addtional suffix (like `<ComponentName>Props`) have to be used to distinguish between them.

For example, I came up to this type structure in my app:
```typescript
export declare namespace BottomSheetTypes {
  export interface Tag {}

  namespace Events {
    interface TagSelectionChanged {
      onSelectionChanged?: (data: Pick<Tag, 'id' | 'isSelected'>) => void;
    }
    ...
  }

  namespace Properties {
    type Tag = {
      id: number;
      isSelected: boolean;
      iconName: IconName;
      iconColor: string;
      label: string;
    } & Events.TagEvents;
  }
}
```

I could do this only using namespaces.

What about adding **type namespaces** which will only allow to use types / interfaces inside them?They can also be nested to better structure type hierarchy.
```typescript
// file1.ts
export type namespace BottomSheetTypes {
  ...
}

// file2.ts
import type { BottomSheetTypes } from "./file1";
```
